tests/test_stackexchange.py

```diff
@@ -0,0 +1,332 @@
+        }
+
+        #unixtime = 1459900800
+        from_date = datetime.datetime(2016, 4, 5)
+        stack = StackExchange(site="stackoverflow", tagged="python", token="aaa", max_questions=1)
+        questions = [question for question in stack.fetch(from_date=from_date)]
+
+        #The date on the questions must be greater than from_date
+        self.assertGreater(questions[0]['updated_on'], 1459900800)
+        for key in expected:
+            self.assertEqual(questions[0][key], expected[key])
+
+        data = json.loads(question)
+        self.assertDictEqual(questions[0]['data'], data['items'][0])
+
+class TestStackExchangeBackendCache(unittest.TestCase):
+    """StackExchange backend tests using a cache"""
+
+    def setUp(self):
+        self.tmp_path = tempfile.mkdtemp(prefix='perceval_')
+
+    def tearDown(self):
+        shutil.rmtree(self.tmp_path)
+
+    @httpretty.activate
+    def test_fetch_from_cache(self):
+        """ Test whether a list of questions is returned from cache """
+
+        question = read_file('data/stackexchange_question')
+
+        httpretty.register_uri(httpretty.GET,
+                               STACKEXCHANGE_QUESTIONS_URL,
+                               body=question, status=200)
+
+        # First, we fetch the bugs from the server, storing them
+        # in a cache
+        cache = Cache(self.tmp_path)
+        stack = StackExchange(site="stackoverflow", tagged="python", token="aaa", max_questions=1, cache=cache)
+
+        questions = [question for question in stack.fetch(from_date=None)]
+        del questions[0]['timestamp']
+
+        # Now, we get the bugs from the cache.
+        # The contents should be the same and there won't be
+        # any new request to the server
+        cache_questions = [cache_question for cache_question in stack.fetch_from_cache()]
+        del cache_questions[0]['timestamp']
+
+        self.assertEqual(cache_questions, questions)
+
+    def test_fetch_from_empty_cache(self):
+        """Test if there are not any questions returned when the cache is empty"""
+
+        cache = Cache(self.tmp_path)
+        stack = StackExchange(site="stackoverflow", tagged="python", token="aaa", max_questions=1, cache=cache)
+
+        cache_questions = [cache_question for cache_question in stack.fetch_from_cache()]
+
+        self.assertEqual(len(cache_questions), 0)
+
+    def test_fetch_from_non_set_cache(self):
+        """Test if a error is raised when the cache was not set"""
+
+        stack = StackExchange(site="stackoverflow", tagged="python", token="aaa", max_questions=1)
+
+        with self.assertRaises(CacheError):
+            _ = [cache_question for cache_question in stack.fetch_from_cache()]
+
+class TestStackExchangeBackendParsers(unittest.TestCase):
+    """StackExchange backend parsers tests"""
+
+    def test_parse_questions(self):
+        """Test question parsing"""
+
+        raw_parse = read_file('data/stackexchange_question')
```

Comment:
This test should parse more that a question. Please add a file that contains more than a question.